ext/sockets/sockets.c

```diff
@@ -1942,11 +1939,20 @@ PHP_FUNCTION(socket_set_option)
 #ifdef TCP_FUNCTION_BLK
 		case TCP_FUNCTION_BLK: {
 			if (Z_TYPE_P(arg4) != IS_STRING) {
-				php_error_docref(NULL, E_WARNING, "Invalid tcp stack name argument type");
-				RETURN_FALSE;
+				zend_argument_type_error(4, "must be of type string when argument #3 ($option) is TCP_FUNCTION_BLK, %s given", zend_zval_value_name(arg4));
+				RETURN_THROWS();
+			}
+			if (zend_str_has_nul_byte(Z_STR_P(arg4))) {
+				zend_argument_value_error(4, "must not contain null bytes when argument #3 ($option) is TCP_FUNCTION_BLK");
+				RETURN_THROWS();
+			}
+			if (Z_STRLEN_P(arg4) > TCP_FUNCTION_NAME_LEN_MAX) {
```

Comment:
might be tired, but should not it be `if (Z_STRLEN_P(arg4) >= TCP_FUNCTION_NAME_LEN_MAX)` ? the error message is correct though.

Reply:
Can it accommodate the trailing null byte to terminate the string? I can't really test this locally :/

Reply:
Just [double checked](https://github.com/freebsd/freebsd-src/blob/da2c88dfcf4f425e6e0a58d6df3a7c8e88d8df92/share/man/man9/tcp_functions.9#L193).